rqt_gui_py/src/rqt_gui_py/rclpy_spinner.py

```diff
@@ -29,7 +29,19 @@ def run(self):
         executor = MultiThreadedExecutor()
         executor.add_node(self._node)
         while rclpy.ok() and not self._abort:
-            executor.spin_once(timeout_sec=1.0)
+            # we have to put a try/except block around the executor
+            # because when ctrl-c there might be the case of waiting within the executor
+            # even though rcl shutdown() was triggered already.
+            # see ticket here: https://github.com/ros-visualization/rqt/issues/198
+            try:
+                executor.spin_once(timeout_sec=1.0)
+            except RuntimeError as re:
```

Comment:
I'm a starting to dig in `rclpy` signal handling related problems.
Could you try the following?
```python3
try:
    executor.add_node(self._node)
    while executor.context.ok() and not self._abort:
        executor.spin_once()
finally:
    executor.remove_node(node)
```
(replacing lines 31-44 of the patch)

I don't have a mac available to try it out.

Reply:
unfortunately, this didn't help and resulted in the same tracebook. Which makes sense as I believe the exception emerges from the `spin_once()` call.

Reply:
Ok. The error is reproducible in Ubuntu too, I will try to take a look.